src/nupic/bindings/algorithms.i

```diff
@@ -101,9 +101,20 @@ _ALGORITHMS = _algorithms
 #include <nupic/proto/ConnectionsProto.capnp.h>
 #include <nupic/proto/SpatialPoolerProto.capnp.h>
 #include <nupic/proto/TemporalMemoryProto.capnp.h>
+%}
 
+//
+// Numpy API
+//
+%{
 #define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION
 #include <numpy/arrayobject.h>
+%}
+%init %{
+  import_array();
```

Comment:
This has always been necessary, as far as I know. A lot of numpy APIs are just macros that don't call into `multiarray.so`, so they don't require this line. But `PyArray_EquivTypenums` calls into `multiarray.so`, so it requires this.